### DISCUSSION TITLE
Possibility of use complex workflows

### DISCUSSION THREAD

**Discussion by fguisso** [projectdiscovery/nuclei-templates#848] [Q&A] [answered]
Hello guys, there is a form to create a workflow with various steps using data from de last step?

e.g. I want to create a workflow that GET in domain content, search for `src=` URLs, then execute a new step in each URLs that a get in the last step.

another e.g. receive a domain and generate new paths based on a waybackurls search and use in the next step.

I ask that because all Workflow found in the repo is about running various templates, and this for my is more a BIG template than a workflow.

**Comment by ehsandeep** [comment, marked as answer]
Hi @fabaff, 

Yes, as of now workflows are limited to run generic and conditional templates, what you asking is more aligned with the issue we are currently working on, GH for tracking https://github.com/projectdiscovery/nuclei/issues/533, this will allow reusing data from 1st request to make new request reusing data from the previous request.

let me know if this is what you are trying to achieve or more example will help to understand you use case?

**Reply by fguisso** [reply]
Good, probably this PR adds the feature, I'll test the dev branch simulate my case, thanks
